lib/hooks/session/index.js

```diff
@@ -187,6 +187,29 @@ module.exports = function(app) {
       // Expose hook as `sails.session`
       app.session = SessionHook;
 
+      // If using Redis, wait for a connection or error
+      if (app.config.session.adapter === 'connect-redis' &&
+          // Future-proof by ensuring that there is a client object, and that it is an event emitter
+          sails.config.session.store.client &&
+          sails.config.session.store.client instanceof require('events').EventEmitter) {
+        return (function() {
+          // If the client is already connected (because it was provided in the config)
+          // then just continue
+          if (sails.config.session.store.client.connected) {return cb();}
+          // Define an error handler for the Redis client which will stop Sails from lifting
+          // by returning an error in the hook callback
+          var redisErrorHandler = function(err) {
+            return cb(err);
+          };
+          sails.config.session.store.client.on('error', redisErrorHandler);
```

Comment:
Actually, just noticed: we'll have to put in a spinlock here as well.  As a basis, here's an implementation of this I recently put together: https://github.com/mikermcneil/machinepack-redis/blob/f4dc7087df27694d5c723784e99a116213d4910e/machines/get-connection.js#L98 

(^^ I hand-tested that code against redis crashes/shutdowns that occur during and after connection, so should be good to go)